packages/lib-utils/src/types/api-discovery.ts

```diff
@@ -0,0 +1,19 @@
+import type { Store, AnyAction } from 'redux';
+import type { ActionType as Action } from 'typesafe-actions';
+import type { K8sVerb } from './k8s';
+
+export type InitAPIDiscovery = (store: Store<unknown, Action<AnyAction>>) => void;
+
+export type APIResourceList = {
```

Comment:
```suggestion
export type APIResourceList = K8sModelCommon & {
```
I wonder if not basing this and dropping `groupVersion` is not better. That way we tie it back to the common model type.

Reply:
This may change some of the logic in the api discovery reducers/actions, but I'm open to this change. I will change it.

Reply:
hmmm... maybe this is not a model then? Did I misunderstand what it was?

Reply:
It is a model, though I can't guarantee what we get back from API discovery will cleanly map into this structure. I'll try it out, but for now I'm fine changing it.